mirascope/anthropic/types.py

```diff
@@ -11,66 +11,133 @@
+from ..base.types import BaseCallParams, BaseCallResponse, BaseCallResponseChunk
 
 
-class AnthropicCompletion(BaseModel):
-    '''Convenience wrapper around Anthropic chat completions.
+class AnthropicCallParams(BaseCallParams[Any]):
+    """The parameters to use when calling d Claud API with a prompt.
+
+    Example:
+
+    ```python
+    from mirascope.anthropic import AnthropicCall, AnthropicCallParams
+
+
+    class BookRecommender(AnthropicCall):
+        prompt_template = "Please recommend some books."
+
+        call_params = AnthropicCallParams(
+            model="anthropic-3-opus-20240229",
+        )
+    ```
+    """
+
+    # TOOLS ARE NOT YET SUPPORTED
+    tools: Annotated[InstanceOf[None], BeforeValidator(lambda x: None)] = None
+
+    max_tokens: int = 1000
+    model: str = "claude-3-sonnet-20240229"
+    metadata: Optional[Metadata] = None
+    stop_sequences: Optional[list[str]] = None
+    system: Optional[str] = None
+    temperature: Optional[float] = None
+    top_k: Optional[int] = None
+    top_p: Optional[float] = None
+    extra_headers: Optional[Headers] = None
+    extra_query: Optional[Query] = None
+    extra_body: Optional[Body] = None
+    timeout: Optional[Union[float, Timeout]] = 600
+
+    wrapper: Optional[Callable[[Anthropic], Anthropic]] = None
+    wrapper_async: Optional[Callable[[AsyncAnthropic], AsyncAnthropic]] = None
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    def kwargs(
+        self, tool_type: Optional[Any] = None, exclude: Optional[set[str]] = None
+    ) -> dict[str, Any]:
+        """Returns the keyword argument call parameters."""
+        extra_exclude = {"wrapper", "wrapper_async"}
+        exclude = extra_exclude if exclude is None else exclude.union(extra_exclude)
+        return super().kwargs(tool_type, exclude)
+
+
+class AnthropicCallResponse(BaseCallResponse[Message, Any]):
+    """Convenience wrapper around the Anthropic Claude API.
 
     When using Mirascope's convenience wrappers to interact with Anthropic models via
-    `AnthropicPrompt`, responses using `AnthropicPrompt.create()` will return a
-    `AnthropicCompletion`, whereby the implemented properties allow for simpler syntax
+    `AnthropicCall`, responses using `Anthropic.call()` will return an
+    `AnthropicCallResponse`, whereby the implemented properties allow for simpler syntax
     and a convenient developer experience.
 
     Example:
 
     ```python
-    from mirascope.anthropic import AnthropicPrompt
+    from mirascope.anthropic import AnthropicCall
 
 
-    class BookRecommendation(AnthropicPrompt):
-        """Please recommend some books."""
+    class BookRecommender(AnthropicCall):
+        prompt_template = "Please recommend some books."
 
 
-    print(BookRecommendation().create())
+    print(BookRecommender().call())
     ```
-    '''
+    """
+
+    # TOOLS NOT YET SUPPORTED
+    tool_types: Annotated[InstanceOf[None], BeforeValidator(lambda x: None)] = None
```

Comment:
tool_types is already defined in BaseCallResponse, is this necessary?

Reply:
This is forcing it to be None since we don't support it.

Reply:
gotcha